libs/utils/src/seqwait.rs

```diff
@@ -43,9 +43,23 @@ where
 {
     waiters: BinaryHeap<Waiter<V>>,
     current: S,
+    status_channel: Sender<Option<V>>,
```

Comment:
nit: would be more readable if type was qualified, i.e.

```suggestion
    status_channel: watch::Sender<Option<V>>,
```